app/models/service_orchestration.rb

```diff
@@ -156,4 +177,28 @@ def save_create_options
                    :create_options      => dup_and_process_password(stack_options))
     save!
   end
+
+  def deploy_stack_job
+    @deploy_stack_job ||= Job.find_by(:id => options.fetch_path(:deploy_stack_job_id))
+  end
+
+  def update_stack_job
+    @update_stack_job ||= Job.find_by(:id => options.fetch_path(:update_stack_job_id))
+  end
+
+  def orchestration_runner_job
+    update_stack_job || deploy_stack_job
+  end
+
+  def wait_on_orchestration_stack
+    while orchestration_runner_job.orchestration_stack.blank?
```

Comment:
clearly this only applies to `deploy_stack_job`. `update_stack_job` does not need to wait on the stack.